### PR TITLE
[blur] Delete prop-types

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- Required for `react-native-web@0.12` https://github.com/expo/expo/issues/8064

# How

- Deleted instances of `prop-types`
- Convert to functional components

# Test Plan

- [x] manually check compliance in NCL

### CODE REVIEW

File: packages/expo-blur/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,8 @@
 
 ### 🎉 New features
 
++ Delete `prop-types` in favor of TypeScript. ([#8676](https://github.com/expo/expo/pull/8676) by [@EvanBacon](https://github.com/EvanBacon))
```
Comment (EvanBacon) [inline packages/expo-blur/CHANGELOG.md:9]:
```suggestion
- Delete `prop-types` in favor of TypeScript. ([#8676](https://github.com/expo/expo/pull/8676) by [@EvanBacon](https://github.com/EvanBacon))
```

**Review by sjchmiela — COMMENTED** [review]
Thank you very much for taking care of this! 🥇

File: packages/expo-blur/src/BlurView.android.tsx [hunk 2]
```diff
@@ -1,27 +1,14 @@
...
-  static defaultProps = {
-    tint: 'default' as BlurTint,
-    intensity: 100,
-  };
-
-  render() {
-    const { tint, intensity, ...props } = this.props;
-
+const BlurView = React.forwardRef<View, BlurProps>(
+  ({ tint = 'default', intensity = 100, style, ...props }, ref) => {
```
[the remark this answers is not in the document]
Comment (EvanBacon) [reply packages/expo-blur/src/BlurView.android.tsx:8]: no, but we should add Android support and resolve it in that PR.

File: packages/expo-blur/src/BlurView.ios.tsx [hunk 3]
```diff
@@ -1,44 +1,23 @@
...
-
-  _setNativeRef = (ref: ComponentOrHandle) => {
-    this._root = ref;
-  };
-
-  setNativeProps = nativeProps => {
-    if (this._root) {
-      NativeModulesProxy.ExpoBlurViewManager.updateProps(nativeProps, findNodeHandle(this._root));
-    }
-  };
```
[the remark this answers is not in the document]
Comment (EvanBacon) [reply packages/expo-blur/src/BlurView.ios.tsx:30]: I tried forwarding the event but I kept getting an error: `Unhandled JS Exception: Invariant Violation: Tried to register two views with the same name ViewManagerAdapter_ExpoBlurView` 
```js
React.useImperativeHandle(
      ref,
      () => {
        const view = nativeRef.current;
        return {
          setNativeProps(props) {
            if (view) {
              NativeModulesProxy.ExpoBlurViewManager.updateProps(props, findNodeHandle(view));
              view.setNativeProps(props);
            }
          },
        } as any;
      },
      [nativeRef]
);
```

I'm just going to keep this as a class for now.

**Pull request merged**